lib/dialogue-agent/entity-linking/entity_lookup.js

```diff
@@ -50,10 +52,12 @@ async function lookupEntity(dlg, entityType, entityDisplay, idEntities) {
             const dialgoueResults = dlg.dialogueState.history[0].results.results;
             let entityCandidates = [];
             for (const item of dialgoueResults) {
+                let canonical = item.value.id.display;
+                canonical = canonical.normalize("NFD").replace(/[\u0300-\u036f]/g, "");
```

Comment:
Is this necessary? We'll do edit distance later right, so small changes should be ok.

I'm just wary of "surprise code" that makes you wonder why is this here...

Reply:
This line is intended to remove accents. I added this because if I try to match beyonce I will end up with Mc Beyonce instead of beyoncé. Even though the edit distance is only once character beyonce doesn't match with beyoncé, so I believe the score for beyoncé would be -0.1. But with Mc Beyonce, theres a match with beyonce so the final score would be around 9.7. One alternate solution could be to increase the penalty for edit distance or decrease the reward for exact matches, but I'm not sure if that would cause any other problems. What do you think?

Reply:
We should make the "exact match" be less exact in the entity scoring computation. In general, heuristics like these fit well in the entity scorer, because entity scoring is inherently heuristic, and we have tests to ensure whatever changes we make to the entity scoring retain high precision and high recall. Changes outside the entity scoring are more dangerous, because the next person reading this code will not remember why it's happening, and why are accents unconditionally removed.